assets/js/src/settings/pages/send_with/other/test_sending.tsx

```diff
@@ -1,64 +1,30 @@
 import React from 'react';
-import MailPoet from 'mailpoet';
 import ReactStringReplace from 'react-string-replace';
 
 import HelpTooltip from 'help-tooltip';
 import { GlobalContext } from 'context';
 import { t, onChange } from 'common/functions';
 import { Label, Inputs } from 'settings/components';
-import { useSetting } from 'settings/store/hooks';
+import { useSetting, useAction } from 'settings/store/hooks';
 
 export default function TestSending() {
   const [email, setEmail] = React.useState<string>((window as any).mailpoet_current_user_email);
   const [mailer] = useSetting('mta');
   const [fromAddress] = useSetting('sender', 'address');
   const { notices } = React.useContext<any>(GlobalContext);
-  const sendTestEmail = async (recipient) => {
-    if (!fromAddress) {
-      return notices.error(<p>{t('cantSendEmail')}</p>, { scroll: true, static: true });
-    }
+  const sendTestEmail = useAction('sendTestEmail');
 
-    MailPoet.Modal.loading(true);
-    try {
-      await MailPoet.Ajax.post({
-        api_version: (window as any).mailpoet_api_version,
-        endpoint: 'mailer',
-        action: 'send',
-        data: {
-          mailer,
-          newsletter: {
-            subject: t('testEmailSubject'),
-            body: {
-              html: `<p>${t('testEmailBody')}</p>`,
-              text: t('testEmailBody'),
-            },
-          },
-          subscriber: recipient,
-        },
-      });
-      notices.success(<p>{t('emailSent')}</p>, { scroll: true });
-      trackTestEmailSent(mailer.method, true);
-    } catch (res) {
-      if (res.errors.length > 0) {
-        notices.error(
-          res.errors.map((err) => <p key={err.message}>{err.message}</p>),
-          { scroll: true }
-        );
+  const sendEmail = async () => {
```

Comment:
It's not that easy because I need to use `notices.error` and `notices.success`, and those are only accessible from inside the React component. Please note that I am not accessing/modifying the state of this component after the `await` keyword so we will not have the typical issues of using an async method inside a React component.

Reply:
I am not sure I follow. The difficulty seems quite unclear to me from redux you would do:
- trigger action `START`
- trigger actual sending
- trigger action `FAILED` or `FINISHED` 

you would listen to those actions in a reducer and update the state accordingly.

Here in the component, you would render a notice when you get a `FAILED` or any other state.